src/layers/attention/mask.rs

```diff
@@ -67,3 +72,98 @@ impl AttentionMask {
         })
     }
 }
+
+/// Errors for attention masks.
+#[derive(Debug, Snafu)]
+pub enum InputMaskError {
+    #[snafu(display("Cannot concatenate masks"))]
+    ConcatMasks { source: candle_core::Error },
+
+    #[snafu(display("Attention mask must be 2D, was {}D", n_dims))]
+    InvalidDims { n_dims: usize },
+}
+
+/// Attention mask.
+///
+/// Sequence elements for which the corresponding mask element is set to
+/// `False` are ignored during attention calculation. Guaranteed to be
+/// a 2D array.
+#[derive(Clone, Debug)]
+pub struct InputMask {
```

Comment:
I feel like we could use better names for this and its counterpart. Maybe `AttentionMask2D` and `AttentionMask4D` instead?

Reply:
Yeah, I couldn't find the right names. But I don't like `AttentionMask2D` and `AttentionMask4D` either, since the names don't describe the semantics, only the shape.

Reply:
Yeah, that sounds better.

Reply:
Fixed.